### PR TITLE
MCR-2287 lazy generate mets in IIIF Presentation API

### PR DESCRIPTION
**Pull request by sebhofmann**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2287).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d4139608b335c03c96e90c009eb8dc01f4c0b5c8! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6113-pull_request-1084

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on d4139608b335c03c96e90c009eb8dc01f4c0b5c8! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/6113-pull_request-1084

File: mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java [hunk 1]
```diff
@@ -50,6 +54,9 @@
 
     private static final Logger LOGGER = LogManager.getLogger();
 
+    public static final boolean STORE_METS_ON_GENERATE = MCRConfiguration2
+        .getOrThrow("MCR.Mets.storeMetsOnGenerate", Boolean::parseBoolean);
```
Comment (yagee-de) [inline mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java:58]: Property naming: After "." start with capital letter - if possible. Also "Mets" is redundant: "MCR.METS.StoreXMLOnGenerate" would be better.

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java (hunk 1 above)
Comment (sebhofmann) [reply mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java:58]: Its not a new property. I just copied it from an other class.

**Review by bt1cu — APPROVED** [review]

**Review by yagee-de — APPROVED** [review]

File: mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java (hunk 1 above)
Comment (kkrebs) [reply mycore-mets/src/main/java/org/mycore/mets/iiif/MCRMetsIIIFPresentationImpl.java:58]: I created a new ticket for this:
https://mycore.atlassian.net/browse/MCR-2308

**Pull request merged**